sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/IntervalUtils.scala

```diff
@@ -119,4 +121,32 @@ object IntervalUtils {
       case _: IllegalArgumentException => null
     }
   }
+
+  /**
+   * Gets interval duration
+   *
+   * @param cal - the interval to get duration
+   * @param targetUnit - time units of the result
+   * @param daysPerMonth - the number of days per one month
+   * @return duration in the specified time units
+   */
+  def getDuration(
+      cal: CalendarInterval,
+      targetUnit: TimeUnit,
+      daysPerMonth: Int = 31): Long = {
```

Comment:
where do we specify `daysPerMonth`?

Reply:
Are there any other places we need to get a duration of interval except for stream watermark? If not I think it's simpler to call it `getDurationForWaterwark`, and explain why we pick 1 month = 31 days, according to https://github.com/apache/spark/pull/16304#discussion_r92753590

Reply:
What do you propose is ad-hoc interface which solves current issue only. I don't like it because it makes interval interface not reusable.
- Next time when we need durations in another place, are you going to add `getDurationForAnotherPlace1`?
- If someone will need a duration with 31 days per month for different purpose but not for watermarks, he/she cannot use it even in Structured Streaming.

Reply:
From my understanding, a watermark has type of `TIMESTAMP` semantically but not `INTERVAL`. Probably, it can be even negative. If to rename the method, it could `getDurationOfDelay` or `getDurationOfTimeout` (or following your proposal `getDurationOfDelayForEventTimeWatermark` or `getDurationOfTimeoutForGroupState`)